### DISCUSSION TITLE
Should the "all" configuration have deprecated rules?

### DISCUSSION THREAD

**Discussion by dangrussell** [angular-eslint/angular-eslint#2057] [Q&A] [answered]
If using the `all` configuration, you need to manually turn off the deprecated rules.  Seems like the deprecated rules should not be part of that configuration. Can anyone offer any insight on this?

**Comment by JamesHenry** [comment, marked as answer]
I do get where you are coming from but that would make deprecating a rule a breaking change, which is not something we want. 

Deprecation can be done any time, removal can only be done in a major. Plus a case could be made that "all means all". I provide the `all` config for completeness, but I have personally never understood why folks would want to opaquely opt into every possible preference and not consider rules outside of the recommended set on a case by case basis...

TL;DR I don't think we should change this

**Reply by dangrussell** [reply]
Thank you for the feedback.

It looks like what I suggested here ended up being done for v19:
- https://github.com/angular-eslint/angular-eslint/pull/2112
- https://github.com/angular-eslint/angular-eslint/pull/2113
- https://github.com/angular-eslint/angular-eslint/pull/2114

**Reply by JamesHenry** [reply]
Previously deprecated rules were entirely deleted in the major version change, that's not the same thing as requesting deprecated rules - _while they still exist_ - not be included in the "all" config
